web/src/engine/osk/src/keyboard-layout/oskLayerGroup.ts

```diff
@@ -249,8 +256,35 @@ export default class OSKLayerGroup {
     return null;
   }
 
-  public refreshLayout(computedWidth: number, computedHeight: number) {
-    this.computedWidth = computedWidth;
-    this.computedHeight = computedHeight;
+
+  public refreshLayout(vkbd: VisualKeyboard, layoutParams: LayerLayoutParams) {
+    // Set layer-group copies of relevant computed-size values; they are used by nearest-key
+    // detection.
+    this.computedWidth = layoutParams.keyboardWidth;
+    this.computedHeight = layoutParams.keyboardHeight;
+
+    // Assumption:  this styling value will not change once the keyboard and
+    // related stylesheets are loaded and applied.
+    if(this._heightPadding === undefined) {
+      // Should not trigger a new layout reflow; VisualKeyboard should have made
+      // no further DOM style changes since the last one.
+
+      // For touch-based OSK layouts, kmwosk.css may include top & bottom
+      // padding on the layer-group element.
+      const computedGroupStyle = getComputedStyle(this.element);
```

Comment:
Also of note:  `VisualKeyboard.refreshLayout()` and `.buildDocumentationKeyboard` will not call this method unless the OSK is in the DOM, so we don't need to check for that.

Only just now realized that I left that assumption unstated here.

Reply:
This comment above ^^^ should be in the code, not in a PR comment.